pkg/terraform/invoke.go

```diff
@@ -0,0 +1,56 @@
+package terraform
+
+import (
+	"fmt"
+	"os"
+
+	"get.porter.sh/porter/pkg/exec/builder"
+)
+
+type InvokeOptions struct {
+	Action string
+}
+
+// Invoke runs a custom terraform action
+func (m *Mixin) Invoke(opts InvokeOptions) error {
+	action, err := m.loadAction()
+	if err != nil {
+		return err
+	}
+	step := action.Steps[0]
+
+	if step.LogLevel != "" {
```

Comment:
This logic to set the log level, change the directory and init terraform seems to be shared across all the commands. It would be good to make a single function that they all call.